kapitan/lint.py

```diff
@@ -0,0 +1,113 @@
+            logger.info("\nInventory path is invalid or not provided, skipping class checks\n")
+        else:
+            logger.info("\nChecking for orphan classes in inventory...\n")
+            status_class_checks = lint_unused_classes(inventory_path)
+
+    status_secrets = 0
+    if search_secrets:
+        logger.info("\nChecking for orphan secrets files...\n")
+        status_secrets = lint_orphan_secrets(compiled_path, secrets_path)
+
+    status = status_secrets + status_class_checks
+    if fail_on_warning and status > 0:
+        sys.exit(1)
+
+    return status
+
+
+def lint_orphan_secrets(compiled_path, secrets_path):
+    logger.debug("Find secret paths for " + secrets_path)
+    secrets_paths = set()
+    for path in list_all_paths(secrets_path):
+        if os.path.isfile(path):
+            path = path[len(secrets_path) + 1:]
+            secrets_paths.add(path)
+
+    logger.debug("Collected # of paths: {}".format(len(secrets_paths)))
+    logger.debug("Checking if all secrets are declared in " + compiled_path)
+
+    for path in list_all_paths(compiled_path):
+        if os.path.isfile(path):
+            with open(path, "r") as compiled_file:
+                file_contents = compiled_file.read()
+                for secret_path in list(secrets_paths):
+                    if secret_path in file_contents:
+                        secrets_paths.discard(secret_path)
+
+    status = len(secrets_paths) > 0
+    if status:
+        logger.info("No usage found for the following {} secrets files:\n{}\n".format(len(secrets_paths), pformat(secrets_paths)))
+
+    return status
+
+
+def lint_unused_classes(inventory_path):
+    classes_dir = os.path.join(inventory_path, "classes/")
+    if not os.path.isdir(classes_dir):
+        raise KapitanError("{} is not a valid directory or does not exist".format(classes_dir))
+
+    logger.debug("Find unused classes from {}".format(classes_dir))
+    class_paths = set()
+    for path in list_all_paths(classes_dir):
+        if os.path.isfile(path) and (path.endswith('.yml') or path.endswith('.yaml')):
+            path = path[len(classes_dir):]
+            path = path.replace(".yml", "").replace(".yaml", "").replace("/", ".")
+            class_paths.add(path)
+
+    logger.debug("Collected # of paths: {}".format(len(class_paths)))
+    logger.debug("Checking if all classes are declared in " + classes_dir)
+
+    for path in list_all_paths(inventory_path):
+        if os.path.isfile(path):
+            with open(path, "r") as compiled_file:
+                file_contents = compiled_file.read()
+                for class_path in list(class_paths):
+                    exists = class_path in file_contents
+                    if class_path.endswith(".init"):
+                        exists = (class_path[:-5] in file_contents) or (exists)
+
+                    if exists:
+                        class_paths.discard(class_path)
+
+    status = len(class_paths) > 0
```

Comment:
I find it a bit inconsistent that status is sometimes and int and other times a boolean

Reply:
Noted, I'll rename in `start_lint`.